src/main/java/io/reactivex/internal/functions/ObjectHelper.java

```diff
@@ -128,4 +128,17 @@ public boolean test(Object o1, Object o2) {
             return ObjectHelper.equals(o1, o2);
         }
     }
+
+    /**
+     * Trap null-check attempts on primitives.
+     * @param value the value to check
+     * @param message the message to print
+     * @return the value
+     * @deprecated this method should not be used as there is no need
+     * to check primitives for nullness.
+     */
+    @Deprecated
+    public static long requireNonNull(long value, String message) {
+        throw new InternalError("Null check on a primitive: " + message);
```

Comment:
Probably not the best choice of error type, `InternalError` indicates JVM error, combined with "Null check on a primitive" message might mislead the developer into thinking that something is wrong with bytecode/VM hehe

Reply:
Especially if they run a value types enabled vm ;)